core/selection.js

```diff
@@ -1018,15 +1027,13 @@
 			}
 
 			function nonEditableAscendant( sel ) {
-				if ( CKEDITOR.env.ie ) {
-					var range = sel.getRanges()[ 0 ],
-						ascendant = range ? range.startContainer.getAscendant( function( parent ) {
-								return parent.type == CKEDITOR.NODE_ELEMENT &&
-									( parent.getAttribute( 'contenteditable' ) == 'false' || parent.getAttribute( 'contenteditable' ) == 'true' );
-							}, true ) : null ;
-
-					return range && ascendant.getAttribute( 'contenteditable' ) == 'false' && ascendant;
-				}
+				var range = sel.getRanges()[ 0 ],
+					ascendant = range ? range.startContainer.getAscendant( function( parent ) {
```

Comment:
Actually I find this whole function really weird. We want to get `[contenteditable=false]` ascendant for the range, yet this function does it in super convoluted way… Why can't it be simplified to something like:

```javascript
function nonEditableAscendant( sel ) {
	var range = sel.getRanges()[ 0 ];

	if( !range ) {
		return null;
	}

	return range.startContainer.getAscendant( function( element ) {
		return element.type === CKEDITOR.NODE_ELEMENT && element.getAttribute( 'contenteditable' ) === 'false';
	}, true );
}
```

Should be working perfectly fine.

Reply:
I have similar question when skimming through this code couple months ago. This check
`parent.getAttribute( 'contenteditable' ) == 'false' || parent.getAttribute( 'contenteditable' ) == 'true'` gave me 2 thoughts:
1. this function actually retrieved an ancestor element with attribute 'contentEditable' that is literally defined. getAttribute() would return 'null' if contentEditable is not defined in html attribute, correct? Otherwise, a better approach to attribute is using native element property .isContentEditable. This property is a boolean, and it is inherited from parent element, or can be set if we literally defined attribute 'contentEditable'.
2. the function's name does not reflect its functionality. I'd recommend a small documentation about what the function does. That would provide a context when skimming through codes.

Thank you.

Reply:
ad. 1) That's not really an issue as every non-editable element in CKEditor's instance has `[contenteditable=false]` directly set.
ad. 2) I think that the name is obvious enough to not include the documentation. However I must agree that it could be better, e.g. `getNonEditableAscendant`. As it's used only in this one place, such change should be safe.